yt_idefix/fields.py

```diff
@@ -47,3 +62,266 @@ def setup_fluid_fields(self):
+                ) / np.sqrt(
+                    data.ds.gamma
+                    * (data[("gas", "pressure")] / data[("gas", "density")])
+                )
+
+            self.add_field(
+                ("gas", "Mach"),
+                sampling_type="cell",
+                function=_mach,
+                units="",
+            )
+            self.alias(
+                ("gas", "mach"),
+                ("gas", "Mach"),
+                units="",
+            )
+
+        if (self.ds._dataset_type, "ndens") in self.field_list:
+            self.add_output_field(
+                (self.ds._dataset_type, "ndens"),
+                sampling_type="cell",
+                units=unit_system["length"] ** -3,
+            )
+            self.alias(
+                ("gas", "ndens"),
+                (self.ds._dataset_type, "ndens"),
+                units=unit_system["length"] ** -3,
+            )
+            self.alias(
+                ("gas", "Number Density"),
+                (self.ds._dataset_type, "ndens"),
+                units=unit_system["length"] ** -3,
+            )
+            self.alias(
+                ("gas", "number density"),
+                (self.ds._dataset_type, "ndens"),
+                units=unit_system["length"] ** -3,
+            )
+            self.alias(
+                ("gas", "Number density"),
+                (self.ds._dataset_type, "ndens"),
+                units=unit_system["length"] ** -3,
+            )
+        else:
+
+            def _ndens(field, data):
+                return data["gas", "density"] / (data.ds.mu * mh)
+
+            self.add_field(
+                ("gas", "Number Density"),
+                sampling_type="cell",
+                function=_ndens,
+                units=unit_system["length"] ** -3,
+            )
+            self.alias(
+                ("gas", "number density"),
+                ("gas", "Number Density"),
+                units=unit_system["length"] ** -3,
+            )
+            self.alias(
+                ("gas", "Number density"),
+                ("gas", "Number Density"),
+                units=unit_system["length"] ** -3,
+            )
+
+        def _velMag(field, data):
+            print("PLUTO ", data)
+            return np.sqrt(
+                data[(self.ds._dataset_type, "vx1")] ** 2
+                + data[(self.ds._dataset_type, "vx2")] ** 2
+                + data[(self.ds._dataset_type, "vx3")] ** 2
+            )
+
+        self.add_field(
+            ("gas", "Speed"),
+            sampling_type="cell",
+            function=_velMag,
+            units=unit_system["velocity"],
+        )
+        self.alias(
+            ("gas", "speed"),
+            ("gas", "Speed"),
+            units=unit_system["velocity"],
+        )
+        self.alias(
+            ("gas", "Velocity Magnitude"),
+            ("gas", "Speed"),
+            units=unit_system["velocity"],
+        )
```

Comment:
`("gas", "velocity_magnitude")` is already defined by yt itself.
```suggestion
```

Reply:
Removed these!